python/paddle/incubate/optimizer/functional/lbfgs.py

```diff
@@ -178,16 +178,20 @@ def body(
             shape=[], fill_value=(head - 1).mod(history_size), dtype='int64'
         )
 
-        def cond(i, q):
+        def cond(i, q, ai_vec):
             return i != tail
 
-        def body(i, q):
-            ai_vec[i] = rhok_vec[i] * paddle.dot(sk_vec[i], q)
+        def body(i, q, ai_vec):
+            ai_vec = paddle.static.setitem(
```

Comment:
The API for ·paddle.static· is only used in static graphs, so it is recommended to add a check to determine whether it is static or dynamic grahp. Only static graphs can call this line of code.

Reply:
Done, thanks.